source/opt/feature_manager.cpp

```diff
@@ -30,5 +37,202 @@ void FeatureManager::Analyze(ir::Module* module) {
   }
 }
 
+void FeatureManager::AddCapabilities(ir::Module* module) {
+  std::stack<SpvCapability> capabilities_to_add;
```

Comment:
Make this code data-driven.  The grammar tables encode the capability dependency graph.  See ValidationState_t::RegisterCapability
This will eliminate the entire switch statement below. (!!!)

It seems the IRContext doesn't have an spv_context, which surprises me.   It should in fact have a libspirv::AssemblyGrammar which does a boatload of stuff in a data-driven way.

Reply:
Done.